insights/parsers/jboss_version.py

```diff
@@ -49,27 +61,76 @@ def parse_content(self, content):
             "release": int(release)
         }
 
+    @property
+    def product(self):
+        """string: the version of this running JBoss progress."""
+        return self._parsed["product"]
+
     @property
     def version(self):
         """string: the version of this running JBoss progress."""
-        return self.parsed["version"]
+        return self._parsed["version"]
 
     @property
     def major(self):
         """int: the major version of this running JBoss progress."""
-        return self.parsed["major"]
+        return self._parsed["major"]
 
     @property
     def minor(self):
         """int: the minor version of this running JBoss progress."""
-        return self.parsed["minor"]
+        return self._parsed["minor"]
 
     @property
     def release(self):
         """int: release of this running JBoss progress."""
-        return self.parsed["release"]
+        return self._parsed["release"]
 
     @property
     def code_name(self):
         """string: code name of this running JBoss progress."""
-        return self.parsed["code_name"]
+        return self._parsed["code_name"]
+
+
+@parser(Specs.jboss_runtime_versions)
+class JbossRuntimeVersions(Parser):
+    """
+     This class is to access to file ``data/insights_commands/jboss_versions``
+
+     Typical content of file ``data/insights_commands/jboss_versions`` is::
+
+         {"/opt/jboss-datagrid-7.3.0-server": "Red Hat Data Grid - Version 7.3.0"}
+
+     This class parses the file content and stores data in the dict ``self.versions``.
+
+     Examples:
+         >>> len(all_jboss_versions)
+         1
+         >>> all_jboss_versions[0].major
+         7
+         >>> all_jboss_versions[0].minor
+         3
+         >>> all_jboss_versions[0].release
+         0
+     """
+
+    def parse_content(self, content):
+        self._versions = []
+        jboss_version_dict = json.loads(' '.join(content))
+        for j_path, version_content in jboss_version_dict.items():
+            self._versions.append(JbossVersion(context_wrap(version_content, path=j_path)))
```

Comment:
The content is {jboss_path: content of version.txt} json. It is no meaningful to use JSONParser to parser the json as the main part is content of version.txt. The class JbossRuntimeVersions stands for running versions of JBoss. So current codes are reflect the logic.  I updated the code to use Context class instead of tests/context_wrap.

Reply:
Updated